### ISSUE TITLE
Add support for running assembly programs on TI-84 Plus CE in OS 5.6+ using arTIfiCE

### ISSUE THREAD

**Issue by JLP04** [debrouxl/tilp_and_gfm#45]
This is more of a feature idea but I was wondering if it would be possible to make it so that if the user has arTIfiCE (https://yvantt.github.io/arTIfiCE/), which allows you to run assembly programs on OS 5.6 and newer, then that could be used to run the assembly scripts for actions like syncing the calculator time and the system time or dumping the ROM through tilp.

**Comment by adriweb** [comment]
Heh, I'm not even sure if keystrokes could still be sent over USB once the app is open, but anyway arTIfiCE should really only be used to install Cesium and/or AsmHook. With the latter, one can just launch Asm programs directly, without the need for tilp or similar to do extra things :)
Although I suppose Tilp could mention such programs...

**Comment by JLP04** [comment]
Yeah, I mostly installed arTIfiCE to dump the ROM to use with CEmu but I was just wondering if it would be possible to implement because it would be convenient to be able to do stuff like sync the time with the computer's time on the newer OSes without having to jailbreak any more than arTIfiCE.

**Comment by JLP04** [comment]
Also, I'm looking at AsmHook, and if you use AsmHook, do you know if there is a way to uninstall it afterward without clearing the RAM?

**Comment by JLP04** [comment]
> Heh, I'm not even sure if keystrokes could still be sent over USB once the app is open, but anyway arTIfiCE should really only be used to install Cesium and/or AsmHook. With the latter, one can just launch Asm programs directly, without the need for tilp or similar to do extra things :)

As of OS 5.8.3, running AsmHook causes the calculator to restart and clear it's RAM. However, arTIfiCE version 2.0 does now work on 5.8.3. Thus, what if there was some way where Tilp could wait a little longer or try a few more times such that it would basically wait for the data to begin being transmitted? Thus, a user could manually start the newly loaded ROMDUMP program using arTIfiCE without having to use the (no longer working) AsmHook.

**Comment by adriweb** [comment]
It's possible to install CeaShell to get asmhook-like functionality, btw.
